packages/realm-react/src/useEmailPasswordAuth.tsx

```diff
@@ -0,0 +1,190 @@
+
+  /**
+   * Complete resetting a user's password.
+   *
+   * @returns `void`
+   */
+  resetPassword(args: { token: string; tokenId: string; password: string }): Promise<void>;
+
+  /**
+   * Call the configured password reset function, passing in any additional
+   * arguments to the function.
+   *
+   * @returns `void`
+   */
+  callResetPasswordFunction<Args extends unknown[] = []>(
+    args: {
+      email: string;
+      password: string;
+    },
+    ...restArgs: Args
+  ): Promise<void>;
+
+  /**
+   * Log out the current user.
+   */
+  logOut(): Promise<void>;
+
+  /**
+   * The {@link AuthResult} of the current (or last) operation performed for
+   * this hook.
+   */
+  result: AuthResult;
+}
+
+export function useEmailPasswordAuth(): UseEmailPasswordAuth {
+  const app = useApp();
+  const [result] = useAuthResult();
+
+  const logIn = useAuthOperation<[{ email: string; password: string }], User>({
+    operation: (args) => app.logIn(Credentials.emailPassword(args)),
+    operationName: AuthOperationName.LogIn,
+  });
+
+  const register = useAuthOperation<[{ email: string; password: string; loginAfterRegister?: boolean }], void>({
+    operation: ({ email, password }) => app.emailPasswordAuth.registerUser({ email, password }),
+    operationName: AuthOperationName.Register,
+    onSuccess: ({ email, password, loginAfterRegister = true }) => {
+      if (loginAfterRegister === true) {
+        return logIn({ email, password });
+      }
+    },
+  });
+
+  const confirm = useAuthOperation({
+    operation: ({ token, tokenId }) => app.emailPasswordAuth.confirmUser({ token, tokenId }),
+    operationName: AuthOperationName.Confirm,
+  });
+
+  const resendConfirmationEmail = useAuthOperation({
+    operation: ({ email }) => app.emailPasswordAuth.resendConfirmationEmail({ email }),
+    operationName: AuthOperationName.ResendConfirmationEmail,
+  });
+
+  const retryCustomConfirmation = useAuthOperation({
+    operation: ({ email }) => app.emailPasswordAuth.retryCustomConfirmation({ email }),
+    operationName: AuthOperationName.RetryCustomConfirmation,
+  });
+
+  const sendResetPasswordEmail = useAuthOperation({
+    operation: ({ email }) => app.emailPasswordAuth.sendResetPasswordEmail({ email }),
+    operationName: AuthOperationName.SendResetPasswordEmail,
+  });
+
+  const callResetPasswordFunction = useAuthOperation({
+    operation: ({ email, password }, ...restArgs) =>
+      app.emailPasswordAuth.callResetPasswordFunction({ email, password }, ...restArgs),
+    operationName: AuthOperationName.CallResetPasswordFunction,
+  });
+
+  const resetPassword = useAuthOperation({
+    operation: ({ password, token, tokenId }) => app.emailPasswordAuth.resetPassword({ password, token, tokenId }),
+    operationName: AuthOperationName.ResetPassword,
+  });
+
+  const logOut = useAuthOperation({
+    operation: () => (app.currentUser ? app.currentUser.logOut() : Promise.resolve()),
```

Comment:
Same suggestion a my earlier one.
```suggestion
    operation: async () => app.currentUser?.logOut(),
```